### PR TITLE
Java methods to sign in user and acquire IdToken from Cognito for Karate

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

File: src/main/java/nva/api/testing/AuthenticationMethods.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            adminSetUserPassword(client, username, password, userPoolId);
+            AdminInitiateAuthResponse authResponse = adminInitiateAuth(
+                    client,
+                    username,
+                    password,
+                    clientId,
+                    userPoolId
+            );
+
+            return authResponse.authenticationResult().idToken();
```
Comment (axthosarouris) [inline src/main/java/nva/api/testing/AuthenticationMethods.java:48]: put this into a private method

File: src/test/java/tests/TestsRunner.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import com.intuit.karate.Results;
+import com.intuit.karate.Runner;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TestsRunner {
+
+    @Test
+    void test() {
```
Comment (axthosarouris) [inline src/test/java/tests/TestsRunner.java]: if this test is supposed to run all tests name it runAllCucumberTests.

File: src/test/java/tests/customers/CustomersRunner.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import com.intuit.karate.Results;
+import com.intuit.karate.Runner;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CustomersRunner {
+
+    @Test
+    void test() {
```
Comment (axthosarouris) [inline src/test/java/tests/customers/CustomersRunner.java]: same

**Review by axthosarouris — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+test {
+    useJUnitPlatform()
+    systemProperty "karate.options", System.properties.getProperty("karate.options")
+    systemProperty "karate.env", System.properties.getProperty("karate.env")
+    outputs.upToDateWhen { false }
+}
+
+repositories {
+    mavenCentral()
+    // mavenLocal()
```
Comment (brinxmat) [inline build.gradle]: ?

File: src/test/java/karate-config.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+function setup() {
+  let authentication = Java.type('nva.api.testing.AuthenticationMethods');
```
Comment (brinxmat) [inline src/test/java/karate-config.js]:
```
function setup() {
  const authentication = Java.type('nva.api.testing.AuthenticationMethods')
  const username = karate.properties.username
  const client_id = karate.properties.clientId
  const user_pool_id = karate.properties.userPoolId
  return {
    BEARER_TOKEN: authentication.getIdToken(username, client_id, user_pool_id),
    SERVER_URL: karate.properties.serverUrl
  }
}
```

**Review by brinxmat — CHANGES_REQUESTED** [review]
Can you also remove the .vscode directory, it's included in .gitignore, but has been committed previously.

Is karate.jar needed, or is this covered in gradle's deps?

Update the github workflow — the directory `tests` has moved to `src/test/java/tests`, I guess

File: src/main/java/nva/api/testing/AuthenticationMethods.java (hunk 1 above)
Comment (janbj) [reply src/main/java/nva/api/testing/AuthenticationMethods.java:48]: @axthosarouris what part of code?

**Review by axthosarouris — APPROVED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**